### DISCUSSION TITLE
What are "Tier 2+ platforms (Cygwin, etc)"?

### DISCUSSION THREAD

**Discussion by MoltenMonster** [fish-shell/fish-shell#10239] [Q&A] [answered]
The [RIIR page](https://github.com/fish-shell/fish-shell/blob/master/doc_internal/fish-riir-plan.md) says "As of yet unknown compatibility story for Tier 2+ platforms (Cygwin, etc)."

**Comment by faho** [comment, marked as answer]
We've never formally defined which operating systems we support, but in practice it works out something like this:

- "Tier 1": Platforms multiple core contributors run as their main system and that are tested in CI - this is Linux (with glibc) and macOS, possibly WSL (no *real* CI, but it's basically linux)
- "Tier 1.5": Platforms that have CI and that contributors may sometimes run - this is FreeBSD
- "Tier 2": Everything else that we know fish runs on - this is Cygwin, OpenBSD, NetBSD, Solaris/Illumos/OpenIndiana, Haiku. You can divide this up further if you want - the two BSDs are a heck of a lot closer to FreeBSD and so will probably work better as a consequence than the rest.

So that "Tier 2+" was a bit of a shorthand for "platforms we don't really regularly test".

The term is, I believe, borrowed from [rustc's support list](https://doc.rust-lang.org/nightly/rustc/platform-support.html).

**Reply by zanchey** [reply]
I read it as literally meaning the Rust platform support list.
